src/qibo/models/circuit.py

```diff
@@ -1099,14 +1099,20 @@ def execute(self, initial_state=None, nshots=1000):
             self._final_state = self.compiled.result(state, nshots)
             return self._final_state
         else:
```

Comment:
Since the first branch ends in `return`, you don't need the `else` (I know it was unaffected by this PR)